src/main/java/org/commonjava/util/partyline/FileTree.java

```diff
@@ -327,46 +327,55 @@ boolean tryLock( File file, String ownerName, String label, LockLevel lockLevel,
         while ( end < 1 || System.currentTimeMillis() < end )
         {
             T result = tryLock( realFile, Thread.currentThread().getName(), "Open File for " + ( doOutput ? "output" : "input" ),
-                                           doOutput ? LockLevel.write : read, WAIT_TIMEOUT, unit, ( opLock ) -> {
+                                           doOutput ? LockLevel.write : read, timeout, unit, ( opLock ) -> {
                         FileEntry entry = entryMap.get( realFile.getAbsolutePath() );
-                        if ( entry.file != null )
+                        boolean proceed = false;
+                        synchronized ( entry )
```

Comment:
The weak entry should be GC'ed immediately after the operation completes since the key is a local string var in withOpLock(). Generally, I think we should not depend on GC behavior since it is unpredictable. This is why it confuses me. Looks like, given a file (path), sometimes different operations share same lock (when it is in the map), and other times they use different locks (when it is GC'ed). This is what I meant by inconsistency. Is this what we want? Or am I missing something?